.github/workflows/ci.yml

```diff
@@ -10,57 +10,138 @@ on:
+        ports:
+          - 6379:6379
 
     steps:
-      - uses: actions/checkout@v4
-      - uses: actions/setup-python@v5
+      - name: Checkout Open Forms
+        uses: actions/checkout@v3
+        with:
+          repository: open-formulieren/open-forms
+          path: open-forms
+
+      - name: Checkout StUF-ZDS payments extension
+        uses: actions/checkout@v3
+        with:
+          path: extension
+
+      - name: Set up backend environment
+        uses: maykinmedia/setup-django-backend@v1.1
         with:
-          python-version: ${{ matrix.python }}
+          apt-packages: 'libxml2-dev libxmlsec1-dev libxmlsec1-openssl gettext postgresql-client gdal-bin'
+          python-version: '3.12'
+          optimize-postgres: 'yes'
+          pg-service: 'postgres'
+          setup-node: 'yes'
+          nvmrc-custom-dir: 'open-forms'
+          npm-ci-flags: '--legacy-peer-deps'
+          working-directory: ${{ github.workspace }}/open-forms
 
-      - name: Install dependencies
-        run: pip install tox tox-gh-actions
+      - name: Make symlink in OF to the extension
+        run: |
+          ln -s ${{ github.workspace }}/extension/stuf_zds_payments ${{ github.workspace }}/open-forms/src
 
       - name: Run tests
-        run: tox
+        run: |
+          export OPEN_FORMS_EXTENSIONS=stuf_zds_payments
+
+          python src/manage.py compilemessages
+          coverage run --source=stuf_zds_payments src/manage.py test stuf_zds_payments
+          coverage combine
+          coverage xml -o coverage-extension.xml
         env:
-          PYTHON_VERSION: ${{ matrix.python }}
-          DJANGO: ${{ matrix.django }}
+          DJANGO_SETTINGS_MODULE: openforms.conf.ci
+          SECRET_KEY: dummy
+          DB_USER: postgres
+          DB_PASSWORD: ''
+        working-directory: ${{ github.workspace }}/open-forms
 
       - name: Publish coverage report
-        uses: codecov/codecov-action@v4
+        uses: codecov/codecov-action@v3.1.4
         with:
-          token: ${{ secrets.CODECOV_TOKEN }}
+          root_dir: ${{ github.workspace }}/extension
+          working-directory: ${{ github.workspace }}/open-forms
+          files: ./coverage-extension.xml
+
+  docker_build:
+    name: Build Docker image
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set tag
+        id: vars
+        run: |
+          # Strip git ref prefix from version
+          VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
+
+          # Strip "v" prefix from tag name (if present at all)
+          [[ "${{ github.ref }}" == "refs/tags/"* ]] && VERSION=$(echo $VERSION | sed -e 's/^v//')
+
+          # Use Docker `latest` tag convention
+          [ "$VERSION" == "main" ] && VERSION=latest
+
+          # PRs result in version 'merge' -> transform that into 'latest'
+          [ "$VERSION" == "merge" ] && VERSION=latest
+
+          echo ::set-output name=tag::${VERSION}
```

Comment:
`set-output` has been deprecated for a while, instead, use env files, like [here](https://github.com/open-formulieren/open-forms/blob/6c8ac92262a0e7e7bbcff30730e9c4e1e6b74eff/.github/workflows/ci.yml#L262)

Feel free to also update the other existing extensions if they have the same issue (I assume you copied over that CI configuration).